File: Algorithms/Java/Merge Sorted Array.java
public class Solution {
    public void merge(int A[], int m, int B[], int n) {
		int i = 0, j = 0;
		while (i < m && j < n) {
			if (A[i] > B[j]) {
				for (int k = A.length - 1; k > i; k--) {
					A[k] = A[k - 1];
				}
				A[i] = B[j];
				i++;
				j++;
				m++;
			} else {
				i++;
			}
		}
		while (j < n) {
			A[i] = B[j];
			i++;
			j++;
		}
	}
}

File: Algorithms/Python/Merge-Sorted-Array.py
# Space O(1), Time O(N)


class Solution(object):

    def merge(self, nums1, m, nums2, n):
        """
        :type nums1: List[int]
        :type m: int
        :type nums2: List[int]
        :type n: int
        :rtype: void Do not return anything, modify nums1 in-place instead.
        """
        i, j, last = m - 1, n - 1, m + n - 1
        while i >= 0 and j >= 0:
            if nums1[i] < nums2[j]:
                nums1[last] = nums2[j]
                j -= 1
            else:
                nums1[last] = nums1[i]
                i -= 1
            last -= 1
        while j >= 0:
            nums1[last] = nums2[j]
            j -= 1
            last -= 1
        print nums1, nums2